Src/EB/AMReX_EB_triGeomOps_K.H

```diff
@@ -0,0 +1,303 @@
+#ifndef AMREX_EB_TRIGEOMOPS_K_H_
+#define AMREX_EB_TRIGEOMOPS_K_H_
+
+#include <AMReX.H>
+
+using namespace amrex;
+
+namespace tri_geom_ops
+{
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE Real Distance2(Real P1[3],Real P2[3])
+    {
+        return( (P1[0]-P2[0])*(P1[0]-P2[0]) +
+                (P1[1]-P2[1])*(P1[1]-P2[1]) +
+                (P1[2]-P2[2])*(P1[2]-P2[2]) );
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE Real DotProd(Real v1[3],Real v2[3])
+    {
+        return(v1[0]*v2[0] + v1[1]*v2[1] + v1[2]*v2[2]);
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE Real side_op(Real L1[6],Real L2[6])
+    {
+        return(    L1[0]*L2[4] 
+                +  L1[1]*L2[5] 
+                +  L1[2]*L2[3] 
+                +  L1[3]*L2[2] 
+                +  L1[4]*L2[0] 
+                +  L1[5]*L2[1] );
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE void getvec(Real P1[3],Real P2[3],Real v[3])
+    {
+        v[0]=P2[0]-P1[0];
+        v[1]=P2[1]-P1[1];
+        v[2]=P2[2]-P1[2];
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE void getunitvec(Real v[3],Real vu[3])
+    {
+        Real vmag;
+        vmag=sqrt(v[0]*v[0]+v[1]*v[1]+v[2]*v[2]);
+        vu[0]=v[0]/vmag;
+        vu[1]=v[1]/vmag;
+        vu[2]=v[2]/vmag;
+    }
+    //================================================================================
+    AMREX_GPU_HOST_DEVICE AMREX_INLINE void CrossProd(Real v1[3],Real v2[3],Real v[3])
```

Comment:
We have `amrex::RealVect`, which has some of the operations like dot product and cross product defined.  I wonder if that can be used.  Note that `RealVect` has only two elements in 2D.  Anyway, this is not a high priority.

Reply:
I am not doing this now. Is it ok if I do it in a future PR?

Reply:
Yes.